### PR TITLE
Add save call

### PR DESCRIPTION
**Pull request by phorne-uncharted**
Added the save and save fitted solution calls. They currently export to `D3MOUTPUTDIR/pipelines` and `D3MOUTPUTDIR/fitted_pipelines` respectively using the solution id and fitted solution id as file names.

Saving a fitted pipeline just pickles the model and outputs it currently. Not every primitive can be pickled so some follow on work will need to be done to make sure pipelines can be pickled.

### CODE REVIEW

File: server/server.py [hunk 1]
```diff
@@ -69,6 +72,10 @@ class ServerServicer(core_pb2_grpc.CoreServicer):
 
     UNIMPLEMENTED_MSG = "Ah, ah, ah, you didn't say the magic word"
 
+    # fitted runtimes should be stored to disk, but that can't happen
+    # until we can guarantee that all primitives can be pickled.
+    fitted_runtimes: Dict[str, runtime.Runtime] = {}
```
Comment (cdbethune) [inline server/server.py]: In the initial version of this, they were pickling the pipelines and saving them to the DB as a blob, using the fitted solution ID as the key.  I think the dict was introduced as a workaround when some of the primitives weren't pickling properly.  Is that an advisable course of action to follow once the pickling issues are worked out, or are we better off just writing to file?

File: server/server.py [hunk 2]
```diff
@@ -147,8 +164,15 @@ def ListPrimitives(self, request, context):
     def Hello(self, request, context):
         return self.msg.make_hello_response_message()
 
+    def add_fitted_runtime(self, solution_id, runtime):
```
Comment (cdbethune) [inline server/server.py]: Seems like these methods and those in `Server` don't really fit here, but I'm guessing there really was no good place to move the dict and the associated methods (original design assumed DB storage of pipelines).  Its better than having the dict declared as a global, but I just wanted to verify the reasoning.

File: server/server.py (hunk 1 above)
Comment (phorne-uncharted) [reply server/server.py]: I think we do want to move away from the dictionary approach once we can pickle reliably. But I wouldn't want to conflate the two issues.

File: server/server.py (hunk 2 above)
Comment (phorne-uncharted) [reply server/server.py]: I mainly wanted to move away from global for now. I originally wanted to drop it into the server, but the server is not linked at all to the task manager (as far as I can tell) whereas the servicer is. Once we pickle them, this is not an issue.

**Pull request merged**